Doesn't work

Tried to download both a collection and a file neither downloaded, but files are resolving.

PS C:\Users\Khris\Documents\Git\krakenpublish> npm install https://github.com/Dahlgren/node-steamworkshop/tarball/2476e75099835374eb5b2a6390f679f3e2eb0234
npm ERR! code E404
npm ERR! 404 Not Found - GET https://github.com/Dahlgren/node-steamworkshop/tarball/2476e75099835374eb5b2a6390f679f3e2eb0234
npm ERR! 404 
npm ERR! 404  'https://github.com/Dahlgren/node-steamworkshop/tarball/2476e75099835374eb5b2a6390f679f3e2eb0234' is not in the npm registry.
npm ERR! 404 Your package name is not valid, because 
npm ERR! 404  1. name can only contain URL-friendly characters
npm ERR! 404  2. name can no longer contain capital letters
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Khris\AppData\Roaming\npm-cache\_logs\2020-07-17T20_57_44_750Z-debug.log

--- a/src/workshop.js
+++ b/src/workshop.js
@@ -89,20 +89,29 @@ SteamWorkshop.prototype.queryFiles = function (query, cb) {
 /* Actual file download */
 
 SteamWorkshop.prototype.saveFileToDisk = function (file, folder) {
-  return new Promise(function (resolve, reject) {
-    var f = fs.createWriteStream(path.join(folder, file.filename))
-      .on('error', function (err) {
-        reject(err)
-      })
-      .on('finish', function () {
-        resolve()
-      })
+  var folderPath = path.join(folder, path.dirname(file.filename))
+  var filePath = path.join(folder, file.filename)
 
-    request(file.file_url)
-      .on('error', function (err) {
-        reject(err)
-      })
-      .pipe(f)
+  return new Promise(function (resolve, reject) {
+    fs.mkdir(folderPath, { recursive: true }, function (err) {
+      if (err) {
+        return reject(err)
+      }
+
+      var f = fs.createWriteStream(filePath)
+        .on('error', function (err) {
+          reject(err)
+        })
+        .on('finish', function () {
+          resolve()
+        })
+
+      request(file.file_url)
+        .on('error', function (err) {
+          reject(err)
+        })
+        .pipe(f)
+    })
   })
 }
 

